Newspack/Newspack/Stores/SiteStore.swift

```diff
@@ -96,20 +216,60 @@ extension SiteStore {
         }
 
         CoreDataManager.shared.performOnWriteContext { [weak self] (context) in
+            defer {
+                DispatchQueue.main.async {
+                    onComplete?()
+                }
```

Comment:
Suggestion! Since `async`'s closure is `() -> Void`, you can just send over the `onComplete` one, unmodified. **And** if it happens to be nil, you just avoid messaging the main thread 🔥 


```
if let onComplete = onComplete {
   DispatchQueue.main.async(execute: onComplete)
}
```

Reply:
Nice tip! Thanks!